lib/request_overrider.js

```diff
@@ -492,31 +491,23 @@ function RequestOverrider(req, options, interceptors, remove, cb) {
       // https://github.com/nodejs/node/blob/2e613a9c301165d121b19b86e382860323abc22f/lib/_http_incoming.js#L67
       response.client = response.socket
 
-      // Evaluate functional headers.
-      const evaluatedHeaders = {}
-      Object.entries(response.headers).forEach(([key, value]) => {
-        if (typeof value === 'function') {
-          response.headers[key] = evaluatedHeaders[key] = value(
-            req,
-            response,
-            responseBody
-          )
-        }
-      })
+      response.rawHeaders.push(
+        ...selectDefaultHeaders(
+          response.rawHeaders,
+          interceptor.scope._defaultReplyHeaders
+        )
+      )
 
-      for (
-        let rawHeaderIndex = 0;
-        rawHeaderIndex < response.rawHeaders.length;
-        rawHeaderIndex += 2
-      ) {
-        const key = response.rawHeaders[rawHeaderIndex]
-        const value = response.rawHeaders[rawHeaderIndex + 1]
+      // Evaluate functional headers.
+      for (let i = 1, len = response.rawHeaders.length; i < len; i = i + 2) {
```

Comment:
I don't think this optimization is necessary and it sort of obscures what this is doing.
```suggestion
      for (let i = 1; i < response.rawHeaders.length; i += 2) {
```

Reply:
I was following existing convention, but would be happy to change.

Reply:
I definitely like to not make the perfect the enemy of the good, and make a point to avoid requesting changes on things that were just moved intact. Sorry I didn't realize from the diff that was the case.